File: 1557. Minimum Number of Vertices to Reach All Nodes
/Solution.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> indegree(n, 0), res;
        for (auto e : edges) {
            indegree[e[1]]++;
        }
        for (int i = 0; i < n; i++) {
            if (indegree[i] == 0) res.push_back(i);
        }
        return res;
    }
};

File: 1557. Minimum Number of Vertices to Reach All Nodes
/Solution.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        indegree = [0 for i in range(n)]
        res = []
        for x, y in edges:
            indegree[y] += 1
        for i, x in enumerate(indegree):
            if x == 0: res.append(i)
        return res